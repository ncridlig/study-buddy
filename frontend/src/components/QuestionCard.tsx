import { Paper, Typography } from '@mui/material';
import { Question } from '@/types';

interface QuestionCardProps {
  question: Question;
}

export default function QuestionCard({ question }: QuestionCardProps) {
  return (
    <Paper sx={{ mb: 2, p: 2, '&:hover': { boxShadow: 3 } }}>
      <Typography variant="subtitle1" fontWeight="bold" component="p" gutterBottom>
        Q: {question.q}
      </Typography>
      <Typography variant="body2" color="text.secondary">
        A: {question.a}
      </Typography>
    </Paper>
  );
}