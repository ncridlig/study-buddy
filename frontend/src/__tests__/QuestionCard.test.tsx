// __tests__/QuestionCard.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import QuestionCard from '@/components/QuestionCard';
import '@testing-library/jest-dom';

describe('QuestionCard', () => {
  const mockQuestion = {
    id: 'q1',
    q: 'What is AI?',
    a: 'Artificial Intelligence',
    // ats: ['Artificial Intelligence', 'Advanced Interface'],
  };

  it('renders the question and answer text', () => {
    render(<QuestionCard question={mockQuestion} />);

    expect(screen.getByText(/^Q: What is AI\?/)).toBeInTheDocument();
    expect(screen.getByText(/^A: Artificial Intelligence/)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<QuestionCard question={mockQuestion} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
