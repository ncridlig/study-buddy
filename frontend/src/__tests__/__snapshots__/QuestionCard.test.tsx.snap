// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`QuestionCard matches snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-4tcbm6-MuiPaper-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <p
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-a99ecy-MuiTypography-root"
  >
    Q: 
    What is AI?
  </p>
  <p
    class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
  >
    A: 
    Artificial Intelligence
  </p>
</div>
`;
